Upstream: Yes
From a0aa1e5b69d6ef0311aeea8e4b9a94eae18a1aaf Mon Sep 17 00:00:00 2001
From: =?utf8?q?Ludovic=20Court=C3=A8s?= <ludo@gnu.org>
Date: Mon, 14 Dec 2009 10:59:25 +0100
Subject: [PATCH] Fix test environment issue with ltdl from Libtool 2.2.6b.

Earlier versions of ltdl would look for extensions under $PWD; this
behavior changed in 2.2.6b.

* test-suite/standalone/Makefile.am (TESTS_ENVIRONMENT): Define
  $builddir.

* test-suite/standalone/test-asmobs: Specify the full path to
  `libtest-asmobs', using $builddir.
---
 test-suite/standalone/Makefile.am |    4 +++-
 test-suite/standalone/test-asmobs |    3 ++-
 2 files changed, 5 insertions(+), 2 deletions(-)

diff --git a/test-suite/standalone/Makefile.am b/test-suite/standalone/Makefile.am
index e7cfd82..058ce93 100644
--- a/test-suite/standalone/Makefile.am
+++ b/test-suite/standalone/Makefile.am
@@ -28,7 +28,9 @@ check_SCRIPTS =
 BUILT_SOURCES =
 EXTRA_DIST =
 
-TESTS_ENVIRONMENT = "${top_builddir}/pre-inst-guile-env"
+TESTS_ENVIRONMENT = 						\
+  builddir="$(builddir)"					\
+  "${top_builddir}/pre-inst-guile-env"
 
 test_cflags =					\
   -I$(top_srcdir)/test-suite/standalone		\
diff --git a/test-suite/standalone/test-asmobs b/test-suite/standalone/test-asmobs
index 2ea75d9..9689ab9 100755
--- a/test-suite/standalone/test-asmobs
+++ b/test-suite/standalone/test-asmobs
@@ -2,7 +2,8 @@
 exec guile -q -s "$0" "$@"
 !#
 
-(load-extension "libtest-asmobs" "libtest_asmobs_init")
+(load-extension (string-append (getenv "builddir") "/libtest-asmobs")
+                "libtest_asmobs_init")
 
 (define (test x v)
   (if v
-- 
1.6.5

